docs/docs/lifting-state-up.md

```diff
@@ -0,0 +1,223 @@
+---
+id: lifting-state-up
+title: Lifting State Up
+permalink: docs/lifting-state-up.html
+prev: state-and-lifecycle.html
+next: composition-vs-inheritance.html
+---
+
+Often, several components need to reflect the same changing data. We recommend to lift the shared state up to their closest common ancestor. Let's see how this works in action.
+
+In this section, we will create a temperature calculator that calculates whether the water would boil at a given temperature.
+
+We will start with a component called `BoilingVerdict`. It accepts the `celsius` temperature as a prop, and prints whether it is enough to boil the water:
+
+```js{3,5}
+function BoilingVerdict(props) {
+  if (props.celsius >= 100) {
+    return <p>The water would boil.</p>;
+  } else {
+    return <p>The water would not boil.</p>;
+  }
+}
+```
+
+Next, we will create a component called `Calculator`. It renders an `<input>` that lets you enter the temperature, and keeps its value in `this.state.celsius`.
+
+Additionally it renders the `BoilingVerdict` for the current input value.
+
+```js{5,10,15,19-22}
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleChange = this.handleChange.bind(this);
+    this.state = {celsius: 0};
+  }
+
+  handleChange(e) {
```

Comment:
Class properties are not officially in the JS spec and require the use of either:
- Babel plugin [transform-class-properties](http://babeljs.io/docs/plugins/transform-class-properties/) **OR**
- Bable preset [Stage 2](https://babeljs.io/docs/plugins/preset-stage-2/) or lower (e.g. Stage 0).

Dan added some great information regarding modern JS to [this file](https://github.com/facebook/react/pull/7920/files#diff-c767b00e65c4eaece82fb5dfb273509eR35).